lib/utils.pm

```diff
@@ -1044,15 +1044,30 @@ sub ensure_serialdev_permissions {
 }
 
 =head2 exec_and_insert_password
-Execute a command that ask for a password and insert it.
+
+    exec_and_insert_password($cmd);
+
+ 1. Execute a command that ask for a password.
+ 2. Detects password prompt
+ 3. Insert password and hits enter
+
 =cut
 sub exec_and_insert_password {
-    my $cmd = shift;
+    my ($cmd) = @_;
+    my $hashed_cmd = hashed_string("SR$cmd");
+    wait_serial($self->{serial_term_prompt}, undef, 0, no_regex => 1) if is_serial_terminal();
```

Comment:
@richiejp  - something like this ?

Reply:
you need to use the serial_terminal.pm module and import serial_term_prompt(). Similar to what you are doing for select_virtio_console().